cmake/GzSanitizers.cmake

```diff
@@ -17,7 +17,7 @@
 
 include(CheckCXXSourceCompiles)
 
-set(IGN_SANITIZER ""
+set(GZ_SANITIZER ""
```

Comment:
This is something we expect users to set, so we should tick-tock it (i.e. also accept `IGN_SANITIZER`, but with warnings).

Reply:
https://github.com/gazebosim/gz-cmake/pull/275/commits/84e0509a8895397cf49d00d7ce0fe2534b29512e !